src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java

```diff
@@ -222,35 +222,42 @@
 	@Value("#{'${mosip.mandatory-languages:}'.concat('${mosip.optional-languages:}')}")
 	private String supportedLang;
 
-	public byte[] generateCard(EventModel eventModel) throws Exception {
+	public boolean generateCard(EventModel eventModel) {
 		Map<String, byte[]> byteMap = new HashMap<>();
+		byte[] pdfbytes=null;
 		String decodedCrdential = null;
 		String credential = null;
-		if (eventModel.getEvent().getDataShareUri() == null || eventModel.getEvent().getDataShareUri().isEmpty()) {
-			credential = eventModel.getEvent().getData().get("credential").toString();
-		} else {
-			String dataShareUrl = eventModel.getEvent().getDataShareUri();
-			URI dataShareUri = URI.create(dataShareUrl);
-			credential = restApiClient.getApi(dataShareUri, String.class);
+		boolean isPrinted=false;
+		try {
+			if (eventModel.getEvent().getDataShareUri() == null || eventModel.getEvent().getDataShareUri().isEmpty()) {
+				credential = eventModel.getEvent().getData().get("credential").toString();
+			} else {
+				String dataShareUrl = eventModel.getEvent().getDataShareUri();
+				URI dataShareUri = URI.create(dataShareUrl);
+				credential = restApiClient.getApi(dataShareUri, String.class);
+			}
+			String ecryptionPin = eventModel.getEvent().getData().get("protectionKey").toString();
+			decodedCrdential = cryptoCoreUtil.decrypt(credential);
+			Map proofMap = new HashMap<String, String>();
+			proofMap = (Map) eventModel.getEvent().getData().get("proof");
+			String sign = proofMap.get("signature").toString();
+			pdfbytes = getDocuments(decodedCrdential,
+					eventModel.getEvent().getData().get("credentialType").toString(), ecryptionPin,
+					eventModel.getEvent().getTransactionId(), sign, "UIN", false).get("uinPdf");
+		}catch (Exception e){
+			isPrinted=false;
+			printLogger.error(e.getMessage() , e);
 		}
-		String ecryptionPin = eventModel.getEvent().getData().get("protectionKey").toString();
-		decodedCrdential = cryptoCoreUtil.decrypt(credential);
-		Map proofMap = new HashMap<String, String>();
-		proofMap = (Map) eventModel.getEvent().getData().get("proof");
-		String sign = proofMap.get("signature").toString();
-		byte[] pdfbytes = getDocuments(decodedCrdential,
-				eventModel.getEvent().getData().get("credentialType").toString(), ecryptionPin,
-				eventModel.getEvent().getTransactionId(), getSignature(sign, credential), "UIN", false).get("uinPdf");
-		return pdfbytes;
+		return isPrinted=true;

```

Comment:
?? There is no return inside the catch block. So all the time you will return true.

Reply:
taken care